### ISSUE TITLE
Nicer error message when passing an empty configuration file

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#9757]
As of the latest Collector version, the error we get if passing an empty configuration file is:

```
Error: invalid configuration: no receiver configuration specified in config
2024/03/14 15:13:40 collector server run finished with error: invalid configuration: no receiver configuration specified in config
```

which, while true, is not very useful. We can add an explicit check to report a more meaningful error message.

**Comment by Kimbohlovette** [comment]
Hello @jpkrohling @mx-psi please could you asign this issue to me?

**Comment by Kimbohlovette** [comment]
@TylerHelmuth @jpkrohling please I need help on signing the CLA

**Comment by cforce** [comment]
I got "collector server run finished with error: invalid configuration: empty configuration file"
Nice enough? The close this issue @mx-psi

**Comment by mx-psi** [comment]
Indeed, this was fixed by #9762
